### ISSUE TITLE
Cannot set JSONDecoder on Codable router

### ISSUE THREAD

**Issue by giginet** [IBM-Swift/Kitura#1298]
## Context and Description

I want to set `dateDecodingStrategy` to decode JSON on codable router.
However, using `JSONDecoder` is hardcoded. So I can't customize behaviors of decoders.

https://github.com/IBM-Swift/Kitura/blob/master/Sources/Kitura/RouterRequest.swift#L208

```swift
struct Foo: Codable {
    let date: Date
}

let jsonString = """
{"date": "1970-01-17T10:52:15-08:00"}
"""
let decoder = JSONDecoder()
decoder.dateDecodingStrategy = .iso8601 // I want to set
let foo = try? decoder.decode(Foo.self, from: jsonString.data(using: .utf8)!)
```

```swift
router.post("/memos") { (foo: Foo, respondWith: (Foo?, RequestError?) -> Void) in
    // Cannot parse `Foo` on Codable routers
}
```

**Comment by ianpartridge** [comment]
Thank you for opening this, we are aware of this current limitation. Do you not have control over the JSON you want to POST, though?

**Comment by svanimpe** [comment]
+1 on this issue for me.

I'm parsing JSON from https://iam.bluemix.net/identity/token which returns an expiration date as a number of seconds since 1970. JSONEncoder/Decoder support this strategy, so it would be nice if I could just configure this and have that number converted to a Date.

**Comment by ianpartridge** [comment]
@Andrew-Lees11 is going to look at fixing this.

**Comment by Andrew-Lees11** [comment]
A pull request to solve this issue has been raised [here](https://github.com/IBM-Swift/Kitura/pull/1309). This adds a dictionary of custom encoders and decoders which will be used for codable routes and so you have access you the JSON decoder if you wish to customize it.

**Comment by Andrew-Lees11** [comment]
This PR has been merged and tagged in Kitura 2.5. A blog post describing the new API's is [here](https://developer.ibm.com/swift/2018/09/01/kitura-custom-encoders-and-decoders/)
